src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -173,6 +173,18 @@ private static RedirectionItem[] LoadRedirectionModel(ErrorBuilder errors, strin
                     // Rebase source_path based on redirection definition file path
                     var basedir = Path.GetDirectoryName(fullPath) ?? "";
 
+                    // Give a warning when source-path not specified in .openpublishing.redirection.json
+                    if (fullPath.EndsWith(".openpublishing.redirection.json"))
+                    {
+                        foreach (var item in redirections)
+                        {
+                            if (item.SourcePath.IsDefault)
+                            {
+                                errors.Add(Errors.Redirection.SourcePathNotSpecified(item.RedirectUrl));
+                            }
+                        }
+                    }
+
                     return (
```

Comment:
Convert this into a `foreach` loop, warn on missing `item.SourcePath` to avoid another loop for warnings

Reply:
Done